test/rules/naming/non-state-vars-leading-underscore.js

```diff
@@ -0,0 +1,76 @@
+const assert = require('assert')
+const { processStr } = require('../../../lib/index')
+const { contractWith } = require('../../common/contract-builder')
+
+const config = {
+  rules: { 'non-state-vars-leading-underscore': 'error' },
+}
+
+describe('non-state-vars-leading-underscore', () => {
+  describe('non state vars should have a leading underscore', function () {
+    ;[
+      {
+        description: 'block variable',
+        underscore: contractWith('function foo() public { uint _myVar; }'),
+        noUnderscore: contractWith('function foo() public { uint myVar; }'),
+      },
+      {
+        description: 'return variable',
+        underscore: contractWith('function foo() public returns (uint256 _foo){}'),
+        noUnderscore: contractWith('function foo() public returns (uint256 foo){}'),
+      },
+      {
+        description: 'function param',
+        underscore: contractWith('function foo( uint256 _foo ) public {}'),
+        noUnderscore: contractWith('function foo( uint256 foo ) public {}'),
+      },
+      {
+        description: 'file-level constant',
+        underscore: 'uint256 constant IMPORTANT_VALUE = 420;',
+        noUnderscore: 'uint256 constant _IMPORTANT_VALUE = 420;',
```

Comment:
@0xGorilla this is a case of a non-state variable where this rule does not enforce it starts with an underscore.

Should I modify the rule implementation so this testcase passes?

also, what should we do for contract-level constants? they're not  part of the _state_

(you can look at the test output in CI logs if you need to btw, no need to set up the project)

Reply:
This case is covered by `private-vars-leading-underscore`, I believe we didn't want to duplicate the functionality.
It's more about visibility than anything else in my opinion, the devs should be able to tell if they're reading/writing to the state just by looking at the name of the variable. What do you think?